email-service/src/main/java/no/sikt/nva/email/EmailRequestHandler.java

```diff
@@ -35,14 +46,68 @@ public EmailRequestHandler() {
     }
 
     @Override
-    protected String processInput(EmailRequest input, RequestInfo requestInfo, Context context)
+    protected String processInput(EmailRequest emailRequest, RequestInfo requestInfo, Context context)
         throws ApiGatewayException {
-        logger.info("processing request");
-        return "hello world";
+        validateRequestBody(emailRequest);
+        var sendEmailResult = sendEmail(emailRequest);
+        logger.info(String.format(EMAIL_LOG_INFO_TRACK_ID, sendEmailResult.getMessageId()));
+        return SUCCESS_MESSAGE;
     }
 
     @Override
     protected Integer getSuccessStatusCode(EmailRequest input, String output) {
         return HttpURLConnection.HTTP_OK;
     }
+
+    private SendEmailResult sendEmail(EmailRequest emailRequest) throws EmailException {
+        return attempt(() -> createSesEmailRequest(emailRequest))
+                   .map(amazonSimpleEmailService::sendEmail)
+                   .orElseThrow(this::logFailureAndThrowEmailException);
+    }
+
+    @SuppressWarnings("PMD.InvalidLogMessageFormat")
```

Comment:
update the PMD config instead of suppressing the warning all the time

Reply:
codacy bug is the reason, not that I don't like to get warnings if the logMessage format is incorrect.